### PR TITLE
Prune Outdated Deposits after Chainstart

### PR DESCRIPTION
**Pull request by nisdas**
This prunes pending deposits when a node performs initial sync, to prevent double counting of already processed deposits.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
I don’t understand this. New deposits in the deposit contract are supposed to go into the pending deposits queue

File: beacon-chain/sync/initial-sync/service.go [hunk 1]
```diff
@@ -14,11 +14,14 @@ import (
...
 	"errors"
 	"fmt"
+	"math/big"
 	"runtime/debug"
 	"strings"
 	"sync"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
+
```
Comment (rauljordan) [inline beacon-chain/sync/initial-sync/service.go]: Spacing

File: beacon-chain/sync/querier.go [hunk 2]
```diff
@@ -2,8 +2,11 @@ package sync
 
 import (
 	"context"
+	"math/big"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
```
Comment (rauljordan) [inline beacon-chain/sync/querier.go]: +1

File: beacon-chain/sync/querier_test.go [hunk 3]
```diff
@@ -3,9 +3,12 @@ package sync
 import (
 	"context"
 	"fmt"
+	"math/big"
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
+
```
Comment (prestonvanloon) [inline beacon-chain/sync/querier_test.go]: spacing

File: beacon-chain/sync/initial-sync/service.go [hunk 4]
```diff
@@ -454,6 +463,19 @@ func (s *InitialSync) processState(msg p2p.Message) {
...
+	if err != nil {
+		log.Errorf("Unable to get powchain block %v", err)
+	}
+
+	if !exists {
+		log.Error("Latest ETH1 block doesn't exist in the pow chain")
+		return
+	}
+
+	// removes old deposits
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/service.go]: remove; this comment doesnt add any new info

File: beacon-chain/sync/initial-sync/service.go [hunk 5]
```diff
@@ -454,6 +463,19 @@ func (s *InitialSync) processState(msg p2p.Message) {
 		return
 	}
 
+	exists, blkNum, err := s.powchain.BlockExists(context.Background(), bytesutil.ToBytes32(beaconState.LatestEth1Data.BlockHash32))
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/service.go]:
```suggestion
	exists, blkNum, err := s.powchain.BlockExists(ctx, bytesutil.ToBytes32(beaconState.LatestEth1Data.BlockHash32))
```

**Pull request merged**